ctest-next/templates/test.rs

```diff
@@ -0,0 +1,111 @@
+        }
+    }
+
+    /// Check that the value returned from the Rust and C side in a certain test is equivalent.
+    ///
+    /// Internally it will remember which checks failed and how many tests have been run. This
+    /// method is the same as `check_same` but prints errors in bytes in hex.
+    fn check_same_hex<T: PartialEq + LowerHex + Debug>(rust: T, c: T, attr: &str) {
+        if rust != c {
+            eprintln!("bad {attr}: rust: {rust:?} ({rust:#x}) != c {c:?} ({c:#x})");
+            FAILED.store(true, Ordering::Relaxed);
+        } else {
+            NTESTS.fetch_add(1, Ordering::Relaxed);
+        }
+    }
+
+    {%- for constant in ffi_items.constants() +%}
+    {%- let ty = constant.ty.to_token_stream().to_string() +%}
+    {%- let ident = constant.ident() +%}
+
+    {%- if ty == "* const c_char" +%}
+    // Test that the string constant is the same in both Rust and C.
+    // While fat pointers can't be translated, we instead of * const c_char.
+    pub fn const_{{ ident }}() {
+        extern "C" {
+            fn __test_const_{{ ident }}() -> *const *const u8;
+        }
+        let val = {{ ident }};
+        unsafe {
+            let ptr = *__test_const_{{ ident }}();
+            // c_char can be i8 or u8, so just cast to i8.
+            let val = CStr::from_ptr(ptr.cast::<i8>());
+            let val = val.to_str().expect("const {{ ident }} not utf8");
+            let c = ::std::ffi::CStr::from_ptr(ptr as *const _);
+            let c = c.to_str().expect("const {{ ident }} not utf8");
+            check_same(val, c, "{{ ident }} string");
+        }
+    }
+    {%- else +%}
+    // Test that the value of the constant is the same in both Rust and C.
+    // This performs a byte by byte comparision of the constant value.
+    pub fn const_{{ ident }}() {
+        extern "C" {
+            fn __test_const_{{ ident }}() -> *const {{ ty }};
+        }
+        let val = {{ ident }};
+        unsafe {
+            let ptr1 = ptr::from_ref(&val).cast::<u8>();
+            let ptr2 = __test_const_{{ ident }}().cast::<u8>();
+            let ptr1_bytes = slice::from_raw_parts(ptr1, mem::size_of::<{{ ty }}>());
+            let ptr2_bytes = slice::from_raw_parts(ptr2, mem::size_of::<{{ ty }}>());
+            for (i, (&b1, &b2)) in ptr1_bytes.iter().zip(ptr2_bytes.iter()).enumerate() {
+                // HACK: This may read uninitialized data! We do this because
+                // there isn't a good way to recursively iterate all fields.
+                check_same_hex(b1, b2, &format!("{{ ident }} value at byte {}", i));
+            }
+        }
+    }
+    {%- endif +%}
+    {%- endfor +%}
+}
+
+use generated_tests::*;
+
+fn main() {
+    println!("RUNNING ALL TESTS");
+    run_all();
+    if FAILED.load( std::sync::atomic::Ordering::Relaxed) {
+        panic!("some tests failed");
+    } else {
+        println!("PASSED {} tests", NTESTS.load( std::sync::atomic::Ordering::Relaxed));
+    }
```

Comment:
```suggestion
    if FAILED.load(std::sync::atomic::Ordering::Relaxed) {
        panic!("some tests failed");
    } else {
        println!(
            "PASSED {} tests",
            NTESTS.load(std::sync::atomic::Ordering::Relaxed)
        );
    }
```

Same thing, just making rustfmt happy with the output files